pickup/lib/ipfs.js

```diff
@@ -170,24 +171,28 @@ export class CarFetcher {
    */
   async fetch ({ cid, abortCtl }) {
     const { ipfsApiUrl, maxCarBytes, fetchTimeoutMs, fetchChunkTimeoutMs } = this
+
+    const fetchTimer = debounce(() => abort(FETCH_TOO_SLOW), fetchTimeoutMs)
+    const chunkTimer = debounce(() => abort(CHUNK_TOO_SLOW), fetchChunkTimeoutMs)
+    const clearTimers = () => {
+      fetchTimer.clear()
+      chunkTimer.clear()
+    }
+    function abort (reason) {
+      clearTimers()
+      if (!abortCtl.signal.aborted) {
+        abortCtl.abort(reason)
+      }
+    }
+
+    // start the clock!
+    fetchTimer()
+    chunkTimer()
```

Comment:
start the timers as soon as fetch is called, not as some point later when the stream composer decides to call the streamWatcher fn.